satpy/composites/__init__.py

```diff
@@ -191,8 +212,10 @@ def check_geolocation(self, data_arrays):
         areas = [ds.attrs.get('area') for ds in data_arrays]
         if all(a is None for a in areas):
             return
-        elif any(a is None for a in areas):
-            raise ValueError("Missing 'area' attribute")
+        for ds in data_arrays:
+            if "area" not in ds.attrs:
+                raise AttributeError(f"Dataset {ds.attrs.get('name', ds.name)!s}"
```

Comment:
This just gives the name of the first data array that misses an area, right?dont we want to find out all data arrays missing area?

Reply:
I think it would be helpful to have the full list of dataarrays missing the area attribute.
Maybe something like this could work?
```python
names_of_dataarrays_missing_an_area = [ds.attrs.get('name', ds.name) for ds in data_arrays if "area" not in ds.attrs]
if names_of_dataarrays_missing_an_area:
    raise AttributeError(f"Dataset(s) {names_of_dataarrays_missing_an_area} miss an 'area' attribute")
```